app/src/gecko/java/org/mozilla/focus/web/GeckoViewPrompt.java

```diff
@@ -0,0 +1,888 @@
+                }
+                view.setBackgroundResource(android.R.drawable.editbox_background);
+                view.getBackground().setColorFilter(color, PorterDuff.Mode.MULTIPLY);
+                return view;
+            }
+        };
+
+        adapter.addAll(0xffff4444 /* holo_red_light */,
+                0xffcc0000 /* holo_red_dark */,
+                0xffffbb33 /* holo_orange_light */,
+                0xffff8800 /* holo_orange_dark */,
+                0xff99cc00 /* holo_green_light */,
+                0xff669900 /* holo_green_dark */,
+                0xff33b5e5 /* holo_blue_light */,
+                0xff0099cc /* holo_blue_dark */,
+                0xffaa66cc /* holo_purple */,
+                0xffffffff /* white */,
+                0xffaaaaaa /* lighter_gray */,
+                0xff555555 /* darker_gray */,
+                0xff000000 /* black */);
+
+        final ListView list = new ListView(builder.getContext());
+        list.setAdapter(adapter);
+        builder.setView(list);
+
+        final AlertDialog dialog = createStandardDialog(builder, callback);
+        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(final AdapterView<?> parent, final View v,
+                                    final int position, final long id) {
+                callback.confirm(String.format("#%06x", 0xffffff & adapter.getItem(position)));
+                dialog.dismiss();
+            }
+        });
+        dialog.show();
+    }
+
+    private static Date parseDate(final SimpleDateFormat formatter,
+                                  final String value,
+                                  final boolean defaultToNow) {
+        try {
+            if (value != null && !value.isEmpty()) {
+                return formatter.parse(value);
+            }
+        } catch (final ParseException e) {
+        }
+        return defaultToNow ? new Date() : null;
+    }
+
+    @SuppressWarnings("deprecation")
+    private static void setTimePickerTime(final TimePicker picker, final Calendar cal) {
+        if (Build.VERSION.SDK_INT >= 23) {
+            picker.setHour(cal.get(Calendar.HOUR_OF_DAY));
+            picker.setMinute(cal.get(Calendar.MINUTE));
+        } else {
+            picker.setCurrentHour(cal.get(Calendar.HOUR_OF_DAY));
+            picker.setCurrentMinute(cal.get(Calendar.MINUTE));
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    private static void setCalendarTime(final Calendar cal, final TimePicker picker) {
+        if (Build.VERSION.SDK_INT >= 23) {
+            cal.set(Calendar.HOUR_OF_DAY, picker.getHour());
+            cal.set(Calendar.MINUTE, picker.getMinute());
+        } else {
+            cal.set(Calendar.HOUR_OF_DAY, picker.getCurrentHour());
+            cal.set(Calendar.MINUTE, picker.getCurrentMinute());
+        }
+    }
+
+    public void promptForDateTime(final GeckoSession session, final String title,
```

Comment:
\[PMD\] INFO: The method 'promptForDateTime' has a Standard Cyclomatic Complexity of 23\.